pkgs/tools/security/bitwarden-directory-connector/default.nix

```
@@ -0,0 +1,103 @@
+{
+  lib,
+  buildNpmPackage,
+  electron,
+  fetchFromGitHub,
+  buildPackages,
+  python3,
+  pkg-config,
+  libsecret,
+  nodejs_18,
+}:
+
+let
+  common = { name, npmBuildScript, installPhase }: buildNpmPackage rec {
+    pname = name;
+    version = "2023.10.0";
+    nodejs = nodejs_18;
+
+    src = fetchFromGitHub {
+      owner = "bitwarden";
+      repo = "directory-connector";
+      rev = "v${version}";
+      hash = "sha256-PlOtTh+rpTxAv8ajHBDHZuL7yeeLVpbAfKEDPQlejIg=";
+    };
+
+    postPatch = ''
+      ${lib.getExe buildPackages.jq} 'del(.scripts.preinstall)' package.json > package.json.tmp
+      mv -f package.json{.tmp,}
+
+      substituteInPlace electron-builder.json \
+        --replace-fail '"afterSign": "scripts/notarize.js",' "" \
+        --replace-fail "AppImage" "dir"
+    '';
+
+    npmDepsHash = "sha256-jBAWWY12qeX2EDhUvT3TQpnQvYXRsIilRrXGpVzxYvw=";
+
+    env.ELECTRON_SKIP_BINARY_DOWNLOAD = "1";
+
+    makeCacheWritable = true;
+    inherit npmBuildScript installPhase;
+
+    buildInputs = [
+      libsecret
+    ];
+
+    nativeBuildInputs = [
+      python3
+      pkg-config
+    ];
+
+    meta = with lib; {
+      description = "LDAP connector for Bitwarden";
+      homepage = "https://github.com/bitwarden/directory-connector";
+      license = licenses.gpl3Only;
+      maintainers = with maintainers; [ Silver-Golden SuperSandro2000 ];
+      platforms = platforms.linux;
+      mainProgram = name;
+    };
+  };
+in {
+  bitwarden-directory-connector = common {
+    name = "bitwarden-directory-connector";
+    npmBuildScript = "build:dist";
+    installPhase = ''
+      runHook preInstall
+
+      npm exec electron-builder -- \
+        --dir \
+        -c.electronDist=${electron}/libexec/electron \
+        -c.electronVersion=${electron.version} \
+        -c.npmRebuild=false
+
+      mkdir -p $out/share/bitwarden-directory-connector $out/bin
+      cp -r dist/linux-unpacked/{locales,resources{,.pak}} $out/share/bitwarden-directory-connector
```

```suggestion
      cp -r dist/*-unpacked/{locales,resources{,.pak}} $out/share/bitwarden-directory-connector
```

according to ofborg this is linux-arm64-unpacked on aarch64-linux...

```
buildPhase completed in 1 minutes 24 seconds
Running phase: glibPreInstallPhase
Running phase: installPhase
  • electron-builder  version=24.6.3 os=5.15.145
  • loaded configuration  file=/build/source/electron-builder.json
  • writing effective config  file=dist/builder-effective-config.yaml
  • skipped dependencies rebuild  reason=npmRebuild is set to false
  • packaging       platform=linux arch=arm64 electron=28.1.4 appOutDir=dist/linux-arm64-unpacked
  • copying Electron  source=/nix/store/gny82l0fkhn92sd3wy58f0pagkdykf4c-electron-28.1.4/libexec/electron destination=/build/source/dist/linux-arm64-unpacked
cp: cannot stat 'dist/linux-unpacked/locales': No such file or directory
cp: cannot stat 'dist/linux-unpacked/resources': No such file or directory
cp: cannot stat 'dist/linux-unpacked/resources.pak': No such file or directory
error: builder for '/nix/store/0n3vc3swbp4hkwk9bdg1q9mgcxwa3hla-bitwarden-directory-connector-2023.10.0.drv' failed with exit code 1;
```